go-controller/pkg/ovn/pods.go

```diff
@@ -243,8 +207,19 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) error {
 	var podMac net.HardwareAddr
 	var podCIDR *net.IPNet
 	var args []string
-
+	var addresses string
+	var addrAssigned bool
 	annotation, err := util.UnmarshalPodAnnotation(pod.Annotations)
+
+	handleError := func(nodeName string, ip *net.IP, assigned bool, retErr error) error {
+		if assigned {
+			if err := oc.lsCache.ReleaseIP(nodeName, ip); err != nil {
+				klog.Error(err)
+			}
+		}
+		return retErr
+	}
+
```

Comment:
Not sure why this can't be handled using `defer` function.

```

var addrAssigned bool
var ip net.IP
defer func() {
		if addrAssigned && err != nil && ip != nil {
			if err := oc.lsCache.ReleaseIP(logicalSwitch, ip); err != nil {
				klog.Error(err)
			}
		}
		return retErr
}
```
The above defer function should work since we have all the variables needed by the defer defined by now.

Reply:
You are right, I wasn't thinking straight. Actually we can declare the defer func in the else{} case and not have to worry about addrAssigned at all.